Source/WebKit/Platform/IPC/Encoder.cpp

```diff
@@ -39,27 +39,25 @@
 
 namespace IPC {
 
-static const uint8_t defaultMessageFlags = 0;
+static constexpr uint8_t defaultMessageFlags = 0;
 
-template <typename T>
-static inline bool allocBuffer(T*& buffer, size_t size)
+static inline std::span<uint8_t> allocBuffer(size_t size)
 {
 #if OS(DARWIN)
-    buffer = static_cast<T*>(mmap(0, size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE, -1, 0));
-    return buffer != MAP_FAILED;
+    auto* buffer = static_cast<uint8_t*>(mmap(0, size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE, -1, 0));
+    return buffer != MAP_FAILED ? std::span { buffer, size } : std::span<uint8_t> { };
 #else
-    buffer = static_cast<T*>(fastMalloc(size));
-    return !!buffer;
+    auto* buffer = static_cast<uint8_t*>(fastMalloc(size));
+    return buffer ? std::span { buffer, size } : std::span<uint8_t> { };
```

Comment:
I don't think `fastMalloc` can ever return `nullptr` so we don't need this `?:` or we should switch to `tryFastMalloc`.

Reply:
You're right, I'll drop the ternary.